pkg/cmd/server/kubernetes/node/node.go

```diff
@@ -112,9 +112,18 @@ func (c *NodeConfig) EnsureDocker(docker *dockerutil.Helper) {
 			return
 		}
 	}
-	if err := dockerClient.Ping(); err != nil {
-		glog.Fatalf("Docker could not be reached at %s.  Docker must be installed and running to start containers.\n%v", endpoint, err)
-		return
+	_, isFakeDocker := client.(*dockertools.FakeDockerClient)
+	if isFakeDocker {
+		// If using the fake docker client, ensure that the CgroupDriver for the kubelet matches
+		// the default cgroup driver.
+		c.KubeletServer.CgroupDriver = "cgroupfs"
+	}
+
+	if !isFakeDocker {
+		if err := dockerClient.Ping(); err != nil {
```

Comment:
Why not adding the `Ping` method to the `FakeDockerClient` rather than introducing this hack?

Reply:
We'd still need to know whether we're using it so we can match the cgroupdriver

Reply:
I will open a PR upstream

Reply:
Thanks.